src/System.Diagnostics.Process/src/System/Diagnostics/Process.NonUap.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace System.Diagnostics
+{
+    public partial class Process : IDisposable
+    {
+        public void Kill(bool entireProcessTree)
+        {
+            if (!entireProcessTree)
+            {
+                Kill();
+            }
+            else
+            {
+                EnsureState(State.Associated | State.IsLocal | State.HaveNonExitedId);
+
+                if (IsSelfOrDescendant(GetCurrentProcess()))
+                    throw new InvalidOperationException(SR.KillEntireProcessTree_DisallowedBecauseTreeContainsCallingProcess);
+
+                IEnumerable<Exception> result = KillTree();
+
+                if (result.Any())
+                    throw new AggregateException(SR.KillEntireProcessTree_TerminationIncomplete, result);
+            }
+        }
+
+        private bool IsSelfOrDescendant(Process processOfInterest)
```

Comment:
IsSelfOrParentOf ?

Reply:
Parent suggest imediate parent, descendant suggests the tree - this is the name XPath is using as well

Reply:
I don't mind adding `Of` though

Reply:
Add `Of`--`IsSelfOrDescendantOf`